pkgs/development/python-modules/kaldi-active-grammar/fork.nix

```diff
@@ -13,15 +13,33 @@
 }:
 
 let
-  old-openfst = openfst.overrideAttrs (self: {
+  old-openfst = openfst.overrideAttrs (prev: {
+    version = "kag-unstable-2022-05-06";
+
     src = fetchFromGitHub {
       owner = "kkm000";
       repo = "openfst";
+      # required by https://github.com/daanzu/kaldi-fork-active-grammar/blob/e9c7d0ffca401cf312779d25f2c05a34b41ff696/cmake/third_party/openfst.cmake#L7
       rev = "0bca6e76d24647427356dc242b0adbf3b5f1a8d9";
       sha256 = "1802rr14a03zl1wa5a0x1fa412kcvbgprgkadfj5s6s3agnn11rx";
     };
     buildInputs = [ zlib ];
-  }); in
+  });
+
+  # commit f07a0615ea82351bbddc0799364774cb350ff759 "openblas: 0.3.20 -> 0.3.21"
+  # ... breaks header compatibility with the outdated kaldi-active-grammar tree.
+  old-openblas = openblas.overrideAttrs (prev: rec {
```

Comment:
Weird things might happen if this package were imported alongside `numpy`

Reply:
Action recommended? It's contained to just this derivation so I don't anticipate much fallout.

Reply:
Took a bit of digging but revealed opencv having similar issues:

```
  * Use LAPACK_xxxx to transparently (via predefined lapack macros) deal with pre and post 3.9.1 versions.
  * LAPACK 3.9.1 introduces LAPACK_FORTRAN_STRLEN_END and modifies (through preprocessing) the declarations of the following functions used in opencv
  *        sposv_, dposv_, spotrf_, dpotrf_, sgesdd_, dgesdd_, sgels_, dgels_
  * which end up with an extra parameter.
  * So we also need to preprocess the function calls in opencv coding by prefixing them with LAPACK_.
  * The good news is the preprocessing works fine whatever netlib's LAPACK version.
```

https://fossies.org/linux/opencv/cmake/OpenCVFindLAPACK.cmake

Reply:
Adding

```
    substituteInPlace src/matrix/cblas-wrappers.h \
      --replace stptri_ LAPACK_stptri \
      --replace dtptri_ LAPACK_dtptri \
      --replace sgetrf_ LAPACK_sgetrf \
      --replace dgetrf_ LAPACK_dgetrf \
      --replace sgetri_ LAPACK_sgetri \
      --replace dgetri_ LAPACK_dgetri \
      --replace sgesvd_ LAPACK_sgesvd \
      --replace dgesvd_ LAPACK_dgesvd \
      --replace ssptri_ LAPACK_ssptri \
      --replace dsptri_ LAPACK_dsptri \
      --replace ssptrf_ LAPACK_ssptrf \
      --replace dsptrf_ LAPACK_dsptrf
```

to `postPatch` allows it to build successfully with 0.3.21

Reply:
Same with openfst. We should not merge this in the current state.

Reply:
The openfst one isn't new and I think it's much less likely to produce a collision - numpy is everywhere on the other hand.

Reply:
Pushed @risicle's suggestion; @SuperSandro2000: KAG uses a openfst fork, so it's not the same.

(I tried typing this with the working build and *it* does work, but effort. Spent like two minutes figuring out I need to say "location <500ms> say <200ms> super <500ms> tab" or a longer string)

Reply:
I also tried it before saying that and it does not work. The fork is based on 1.7, we are on 1.8 - I am not interested in debugging this, and it's out of the scope of my changes: cleaning [openfst] up.